swap_meet/vendor.py

```diff
@@ -1,2 +1,66 @@
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if category == item.category:
+                category_list.append(item)
+
+        return category_list
+
+    def swap_items(self, someonelse, my_item, someonelse_item):
+        if my_item in self.inventory and someonelse_item in someonelse.inventory:
+            self.remove(my_item)
+            someonelse.add(my_item)
+            someonelse.remove(someonelse_item)
+            self.add(someonelse_item)
+            return True
+        else:
+            return False
+
+    def swap_first_item(self, someonelse):
+
+        if len(self.inventory) == 0 or len(someonelse.inventory) == 0:
```

Comment:
Nice guard clause!

Is there a way to check that the inventory is non-empty without explicitly using `len`? Using `len` is perfectly fine, but you'll see the version that doesn't use `len` _very_ frequently in other Python code.